src/lib/services/server/kv/index.ts

```diff
@@ -0,0 +1,47 @@
+import { createClient as createRedisClient } from "redis";
+
+export type Client = {
+  getBlurhashByURL: (url: string) => Promise<string | null>;
+  setBlurhashByURL: (url: string, blurhash: string) => Promise<void>;
+};
+
+type None = Record<never, never>;
+
+const keys = {
+  blurhashByURL: "blurhashByURL",
+};
+
+export const createClient = async ({
+  url = "redis://localhost",
+}: { url?: string } = {}): Promise<Client> => {
+  const parsedURL = new URL(url);
+  // "localhost" is the hostname used when running a redis server locally on the dev machine. "kv"
+  // is the hostname used when running in Docker
+  const useTLS = !(parsedURL.hostname === "localhost" || parsedURL.hostname === "kv");
+  let redisClientError: undefined | unknown;
+  const redisClient = createRedisClient<None, None, None>({
+    url,
+    socket: { tls: useTLS },
```

Comment:
I _think_ we may be able to fix the socket connection error with the `pingInterval` config, although I'm not sure if this could have side effects outside of local development (so we may only want to add this config for local dev). After making this change locally, I didn't see the error pop up after running `dev` for over 10 minutes (usually I saw the error pop up around the 5 minute mark).

```suggestion
    socket: { tls: useTLS },
    // ping every 5 minutes to avoid a socket connection error in local development
    pingInterval: 5 * 60 * 1000,
  });
```